### PR TITLE
Moved endpoints to json endpoints, added specs

### LINKED ISSUE #5: Jobs - Create
Create a new job once the user accepts the price for the job

POST	/jobs
	Params:
	{
               from: {address: String},
               to: {address: String},
               notes: string
         }
Returns
	200 - Successful creation
	403 - Unauthorized

### PR DESCRIPTION
**Pull request by jarrydlee**
resolves #5

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -28,6 +20,18 @@
       post '', to: 'reviews#create', as: 'create'
     end
 
+    scope 'jobs', as: 'jobs' do
+      post '', to: 'job#create', as: 'create_job'
+    end
+
+    scope 'price', as: 'price' do
+      post 'price', to: 'price#price', as: 'price'
```
Comment (jarrydlee) [inline config/routes.rb]: I will need to fix these routes before PR is merged

File: app/controllers/job_controller.rb [hunk 2]
```diff
@@ -1,36 +1,31 @@
...
 
   # POST /job
-  def create_post
-    if user_signed_in?
+  def create
 
-      params_price = params['total_price'].to_i
-      params_from = params['from_address']
-      params_to = params['to_address']
+    params_price = params['total_price'].to_i
```
Comment (robmcl4) [inline app/controllers/job_controller.rb]: `to_i` may limit us to only integer prices ... should we use `to_f` instead?

Comment (jarrydlee) [inline app/controllers/job_controller.rb]: Yeah that's a good point, will fix

File: app/controllers/price_controller.rb [hunk 3]
```diff
@@ -1,23 +1,23 @@
...
-      session['to_address'] = @to_address
-      session['from_address'] = @from_address
+    # Store information in session for use later in the process
+    session['price'] = @total_price
+    session['to_address'] = @to_address
+    session['from_address'] = @from_address
 
-      render 'flap_main/price', layout: 'layouts/flap_main'
-    end
+    render(nothing: true, status: 200)
```
Comment (robmcl4) [inline app/controllers/price_controller.rb]: Shouldn't we respond with the price so the frontend knows what the price is? Ex:
```json
{
  "price": 34
}
```

File: app/controllers/transaction_controller.rb [hunk 4]
```diff
@@ -1,11 +1,25 @@
...
   # GET /jobs/:transaction_id
-  def display
-    @transaction = Transaction.find_by_id!(params[:transaction_id])
-    if @transaction.user == current_user || @transaction.driver == current_user
-      render 'flap_main/transaction_display', layout: 'layouts/flap_main'
+  def get_transaction
+    transaction = Transaction.find_by_id(params[:transaction_id])
+    if !transaction.nil? && (transaction.user == current_user || (!transaction.driver.nil? && transaction.driver.user == current_user))
+      render json: {
+        transaction: transaction
```
Comment (robmcl4) [inline app/controllers/transaction_controller.rb]: Does this just work? I feel like it is going to call `.to_s` or `.to_h` on the Transaction object and end up with weird json being sent to the client.

File: app/controllers/price_controller.rb (hunk 3 above)
Comment (jarrydlee) [inline app/controllers/price_controller.rb]: I've also added in the to/from address

File: app/controllers/transaction_controller.rb (hunk 4 above)
Comment (jarrydlee) [inline app/controllers/transaction_controller.rb]: It actually does turn the transaction object into json format and return that

**Pull request merged**